Service/SamlDTO.php

```diff
@@ -137,6 +144,11 @@ public function __construct(array $samlAttributes)
         $this->departmentId = $this->departmentIds[2] ?? null;
         $this->subDepartmentId = $this->departmentIds[3] ?? null;
         $this->officeId = $this->departmentIds[4] ?? null;
+
+        $employeeEmailArray = \explode(';', $this->getAttributeValue(self::EMPLOYEE_LIST_ATTRIBUTE, $samlAttributes));
+        $this->employeeList = \array_filter($employeeEmailArray, function ($value) {
+            return '' !== $value;
+        });
```

Comment:
I would use [`preg_split`](https://www.php.net/manual/en/function.preg-split.php) for this

```suggestion
        $this->employeeList = \preg_split('/\s*;\*/', $this->getAttributeValue(self::EMPLOYEE_LIST_ATTRIBUTE, $samlAttributes), -1, PREG_SPLIT_NO_EMPTY);
```

but that's a matter of habit and preference.

I would also use just `\array_filter($employeeEmailArray)` to weed out empty values, but being explicit makes sense.

However, your solution is perfectly fine and sound.

Reply:
Yes. But `explode()` is [faster](https://www.php.net/manual/en/function.preg-split.php). So if we are micro optimizing then `preg_split()` is the wrong choice. Only advantage is that `preg_split()` can filter empty values. Benchmarking which is faster, `array_filter(explode())` or `preg_split()` is left as an exercise for the reader 😄 

Note: Most array sized expected to be `N < 20`